src/sinol_make/commands/run/__init__.py

```diff
@@ -483,6 +481,14 @@ def calculate_points(self, results):
 
     def compile_and_run(self, solutions):
         compilation_results = self.compile_solutions(solutions)
+        offset = 0
+        for i in range(len(solutions)):
+            if not compilation_results[i]:
+                self.failed_compilations.append(solutions[i - offset])
+                solutions.pop(i - offset)
+                offset += 1
```

Comment:
Just to be sure: this code works when some elements get popped? It's not clear that this is correct. A more readable solution would be to create a second `solutions` list, or something other.